### PR TITLE
Add GELU layer for vision transformers

### PR DESCRIPTION
**Pull request by fengyuentau**
This PR adds the CPU and OCL kernels of GELU and GELU-Approximation layers.

Merge with https://github.com/opencv/opencv_extra/pull/1044

References:

- https://github.com/pytorch/vision/blob/main/torchvision/models/vision_transformer.py
- https://pytorch.org/docs/stable/generated/torch.nn.GELU.html

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/dnn/src/layers/elementwise_layers.cpp [hunk 1]
```diff
@@ -837,6 +837,63 @@ struct BaseDefaultFunctor : public BaseFunctor
...
+    {
+        return backendId == DNN_BACKEND_OPENCV;
+    }
+
+    inline float calculate(float x) const
+    {
+        return 0.5f * x * (1.0f + erf(x * M_SQRT1_2));
+    }
+
+    int64 getFLOPSPerElement() const { return 1; }
```
Comment (alalek) [inline modules/dnn/src/layers/elementwise_layers.cpp:856]: > 1

4 + erf calculation/approximation.
30-100 is more applicable here.

**Review by alalek — COMMENTED** [review]
Looks good!

File: modules/dnn/src/layers/elementwise_layers.cpp (hunk 1 above)
Comment (fengyuentau) [reply modules/dnn/src/layers/elementwise_layers.cpp]: Ok

**Review by alalek — APPROVED** [review]
Thank you!

**Comment by fengyuentau** [comment]
@rogday Could you review if possible?

File: modules/dnn/src/opencl/activations.cl [hunk 2]
```diff
@@ -307,6 +307,22 @@ __kernel void ThresholdedReluForward(const int n, __global T* in, __global T* ou
...
+}
+
+__kernel void GeluApproximationForward(const int n, __global T* in, __global T* out,
+                                       const KERNEL_ARG_DTYPE sqrt_2_pi,
+                                       const KERNEL_ARG_DTYPE coef_sqrt_2_pi)
+{
+  int index = get_global_id(0);
+    if(index < n)
+        out[index] = (T)0.5f * in[index] * ( (T)1.f + tanh(in[index] * (sqrt_2_pi + coef_sqrt_2_pi * in[index] * in[index])) );
+}
```
Comment (alalek) [inline modules/dnn/src/opencl/activations.cl:324]: Please use this OpenCL code:

```
__kernel void GeluForward(const int n, __global T* in, __global T* out)
{
    int index = get_global_id(0);
    if (index < n)
    {
        T x = in[index];
        out[index] = (T)0.5f * x * ( (T)1.f + erf(x * M_SQRT1_2) );
    }
}

__kernel void GeluApproximationForward(const int n, __global T* in, __global T* out)
{
    // see GeluApproximationConstants from .cpp
    const T sqrt_2_pi = 0.7978845834732056f;
    const T coef_sqrt_2_pi = 0.044714998453855515f * sqrt_2_pi;

    int index = get_global_id(0);
    if(index < n)
    {
        T x = in[index];
        out[index] = (T)0.5f * x * ( (T)1.f + tanh(x * (sqrt_2_pi + coef_sqrt_2_pi * x * x)) );
    }
}
```

and drop `setKernelParams()` method.

Comment (fengyuentau) [reply modules/dnn/src/opencl/activations.cl:324]: Okay
